backend/poll.go

```diff
@@ -0,0 +1,186 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"golang.org/x/net/context"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/datastore"
+	"net/http"
+)
+
+const (
+	ERROR_CASE_POLL  = "error"
+	POLL_SAMPLE_PATH = "/" + CATEGORY_SAMPLE_TEMPLATES + "/poll/"
+	POLL_ANSWER      = "PollAnswer"
+)
+
+//holds the answer chosen by the user and the client id, used for storing data coming from the UI
+type PollForm struct {
+	ClientId string
+	Answer   string
+}
+
+//holds an answers and votes represented as an array, used for displaying
+type PollEntryResult struct {
+	Percentage []int
+	Answer     string
+}
+
+//holds the poll results and a message, used for displaying
+type PollResult struct {
```

Comment:
I find it a bit confusing that a struct called PollResult contains a list of PollResults